src/package/abi/Microsoft.Windows.AbiWinRT.targets

```diff
@@ -6,8 +6,493 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
     <PropertyGroup>
-        <AbiWinRTPackageDir Condition="'$(AbiWinRTPackageDir)'==''">$([System.IO.Path]::GetFullPath($(MSBuildThisFileDirectory)))..\..\</AbiWinRTPackageDir>
-        <AbiWinRTPath Condition="'$(AbiWinRTPath)'==''">"$(AbiWinRTPackageDir)bin\"</AbiWinRTPath>
+        <!-- Only do this for MSBuild versions below 16.0
+             as it is since done automatically, see https://github.com/microsoft/msbuild/pull/3605-->
+        <MSBuildAllProjects Condition="'$(MSBuildToolsVersion)'  &lt;= '15'">$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
     </PropertyGroup>
 
-</Project>
+    <PropertyGroup>
+        <AbiWinRTVerbosity Condition="'$(AbiWinRTVerbosity)' == ''">normal</AbiWinRTVerbosity>
+        <AbiWinRTCommandVerbosity Condition="'$(AbiWinRTVerbosity)' == 'high'">-verbose</AbiWinRTCommandVerbosity>
+        <AbiWinRTProjectWinMD>$(OutDir)$(RootNamespace).winmd</AbiWinRTProjectWinMD>
+        <AbiWinRTMergedDir>$(IntDir)Merged\</AbiWinRTMergedDir>
+        <AbiWinRTUnmergedDir>$(IntDir)Unmerged\</AbiWinRTUnmergedDir>
+        <AbiWinRTSkipUnchangedFiles Condition="'$(AbiWinRTSkipUnchangedFiles)' == ''">true</AbiWinRTSkipUnchangedFiles>
+        <AbiWinRTUseHardlinksIfPossible Condition="'$(AbiWinRTUseHardlinksIfPossible)' == ''">false</AbiWinRTUseHardlinksIfPossible>
+        <AbiWinRTPath Condition="'$(AbiWinRTPath)'==''">$(MSBuildThisFileDirectory)..\..\bin\</AbiWinRTPath>
+
+        <!-- Default modern idl semantics to false considering that most users probably have not made the switch to the modern idl syntax -->
+        <AbiWinRTModernIDL Condition="'$(AbiWinRTModernIDL)'==''">false</AbiWinRTModernIDL>
+
+        <!-- Only generate the platform and component projection if using modern idl. Otherwise, allow the legacy
+             toolchain to kick in, as well as use the existing references from the Windows SDK -->
+        <AbiWinRTEnableComponentProjection Condition="'$(AbiWinRTEnableComponentProjection)' == '' and '$(AbiWinRTModernIdl)'!='false'">true</AbiWinRTEnableComponentProjection>
+        <AbiWinRTEnablePlatformProjection Condition="'$(AbiWinRTEnablePlatformProjection)' == '' and '$(AbiWinRTModernIdl)'!='false'">true</AbiWinRTEnablePlatformProjection>
+        <AbiWinRTEnableReferenceProjection Condition="'$(AbiWinRTEnableReferenceProjection)' == '' and '$(AbiWinRTModernIdl)'!='false'">true</AbiWinRTEnableReferenceProjection>
```

Comment:
cppwinrt makes a distinction between platform, component, and reference projections which abi.exe should not do.  there should only be a single combined platform/reference projection here (component is a unique cppwinrt concept implying authoring support, not just consuming a third party lib).  and there should be a build assertion to prevent contamination with SDK abi headers.  maybe a static assert or pragma detect mismatch would help

Reply:
Does the component scenario not make sense for abi.exe since midlrt generates the same headers?

Reply:
how do you prevent contamination with sdk abi headers? we could probably remove the sdk abi header location from `ClCompile.AdditionalIncludeDirectories`?

Reply:
actually I don't think we can do that, since wrl headers are also in that location

Reply:
talk to @manodasanW - he had to tweak the SDK Nuget to prevent collisions with the uap.props additional include dirs

Reply:
The way I did it was by pre-pending `GeneratedFilesDir` to the `ClCompile.AdditionalIncludeDirectories`. I'll follow up with Mano to see what he suggests. I'd love for an actual assert that isn't dependent on MSBuild import order.